### PR TITLE
Point shader improvements

### PR DESCRIPTION
**Pull request by almarklein**
Ticks a few boxes in  #48

* [x] Move point shader into wgsl file.
* [x] `PointMaterial.size_mode` replaces `vertes_sizes` to be consistent with `color_mode`.
* [x]  `PointMaterial.size_space`, similar to line.
* [x] `PointMaterial.aa` to control anti-aliasing behavior.
* [x] Good nan-behavior.
* [x] Validation examples.

Markers -> new pr.

### CODE REVIEW

File: pygfx/materials/_points.py [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
 
 class PointsMaterial(Material):
     """Point default material.
 
-    Renders (antialiased) disks of the given size and color.
+    Renders disks of the given size and color.
 
     Parameters
     ----------
+    size : int
```
Comment (Korijn) [inline pygfx/materials/_points.py:13]: is this a float actually?

Comment (almarklein) [reply pygfx/materials/_points.py:13]: yes

Comment (almarklein) [reply pygfx/materials/_points.py:13]: fixed

**Comment by almarklein** [comment]
By creating the validation examples, I found a bug related to tiny points (smaller than a pixel), that also affected lines. Fixed it in both. 🚀

**Review by Korijn — APPROVED** [review]

**Pull request merged**